dask/highlevelgraph.py

```diff
@@ -540,9 +552,13 @@ class MaterializedLayer(Layer):
         The mapping between keys and tasks, typically a dask graph.
     """
 
-    def __init__(self, mapping: Mapping, annotations=None):
-        super().__init__(annotations=annotations)
+    def __init__(self, mapping: Mapping, annotations=None, collection_annotations=None):
```

Comment:
This materialized layer stuff belongs in a separate PR, imo.
Also, isn't there some extra logic about what to do if no `collection_annotations` are passed in (I seem to remember we looked in the dask config, but maybe that was only for `annotations` instead)

Reply:
The issue with that is that I need all the layers (`Layers` and `MaterializedLayers` both) to have a `collection_annotations` dictionary.

I guess, for now, I can just add an extra if statement up top and remove the changes I made to `MaterializedLayer` class.

```py
svg_repr = ""
if (
    self.collection_annotations is not None
    and self.collection_annotations.get("type") == "dask.array.core.Array"
):
    chunks = self.collection_annotations.get("chunks")
    if chunks:
        from .array.svg import svg

        svg_repr = "<br />" + svg(chunks)
```

This also works :heavy_check_mark: and I feel it's a better method for now. I will commit this :+1: 

According to my knowledge, the config method doesn't seem to do anything for `annotations` and `collection_annotations` as we only have the following keys only.

```py
config = {
            "temporary-directory": None,
            "dataframe": {"shuffle-compression": None},
            "array": {
                "svg": {"size": 120},
                "slicing": {"split-large-chunks": None},
                "chunk-size": "128MiB",
                "rechunk-threshold": 4,
            },
            "optimization": {
                "fuse": {
                    "active": None,
                    "ave-width": 1,
                    "max-width": None,
                    "max-height": inf,
                    "max-depth-new-edges": None,
                    "subgraphs": None,
                    "rename-keys": True,
                }
            },
        }
   ```

Am I missing out on something? I checked around in the `config,py` as well.